internal/core/src/exec/expression/NullExpr.cpp

```diff
@@ -0,0 +1,137 @@
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "NullExpr.h"
+#include <memory>
+#include <utility>
+#include "common/Array.h"
+#include "common/Types.h"
+#include "log/Log.h"
+#include "query/Utils.h"
+namespace milvus {
+namespace exec {
+
+void
+PhyNullExpr::Eval(EvalCtx& context, VectorPtr& result) {
+    switch (expr_->column_.data_type_) {
+        case DataType::BOOL: {
+            result = ExecVisitorImpl<bool>();
+            break;
+        }
+        case DataType::INT8: {
+            result = ExecVisitorImpl<int8_t>();
+            break;
+        }
+        case DataType::INT16: {
+            result = ExecVisitorImpl<int16_t>();
+            break;
+        }
+        case DataType::INT32: {
+            result = ExecVisitorImpl<int32_t>();
+            break;
+        }
+        case DataType::INT64: {
+            result = ExecVisitorImpl<int64_t>();
+            break;
+        }
+        case DataType::FLOAT: {
+            result = ExecVisitorImpl<float>();
+            break;
+        }
+        case DataType::DOUBLE: {
+            result = ExecVisitorImpl<double>();
+            break;
+        }
+        case DataType::VARCHAR: {
+            if (segment_->type() == SegmentType::Growing &&
+                !storage::MmapManager::GetInstance()
+                     .GetMmapConfig()
+                     .growing_enable_mmap) {
+                result = ExecVisitorImpl<std::string>();
+            } else {
+                result = ExecVisitorImpl<std::string_view>();
+            }
+            break;
+        }
+        case DataType::JSON: {
+            result = ExecVisitorImpl<Json>();
+            break;
+        }
+        case DataType::ARRAY: {
+            result = ExecVisitorImpl<ArrayView>();
+            break;
+        }
+        default:
+            PanicInfo(DataTypeInvalid,
+                      "unsupported data type: {}",
+                      expr_->column_.data_type_);
+    }
+}
+
+template <typename T>
+VectorPtr
+PhyNullExpr::ExecVisitorImpl() {
+    if (auto res = PreCheckNullable()) {
+        return res;
+    }
+    TargetBitmap valid_res = ProcessChunksForValid<T>(is_index_mode_);
+    TargetBitmap res = valid_res.clone();
+    if (expr_->op_ == proto::plan::NullExpr_NullOp_IsNull) {
+        res.flip();
+    }
+    auto res_vec =
+        std::make_shared<ColumnVector>(std::move(res), std::move(valid_res));
+    return res_vec;
+}
+
+ColumnVectorPtr
+PhyNullExpr::PreCheckNullable() {
```

Comment:
What does this function do?

Reply:
To check if nullable is false, then the TargetBitmap is all true when is_not_null, and all false when is_null.